pkg/routes/quickstarts.go

```diff
@@ -50,19 +51,56 @@ func findQuickstartsByTags(tagTypes []models.TagType, tagValues [][]string, pagi
 	// Add pagination
 	err := query.Limit(pagination.Limit).Offset(pagination.Offset).Find(&quickstarts).Error
 
+	return quickstarts, err
+}
+
+// Helper function to find the intersection of two Quickstart arrays
+func intersectQuickstarts(a, b []models.Quickstart) []models.Quickstart {
+	intersection := []models.Quickstart{}
+	set := make(map[uint]models.Quickstart)
+
+	// Populate the map with items from the first array (a), using the ID field
+	for _, q := range a {
+		set[q.ID] = q
+	}
+
+	// Check for items in the second array (b) that also exist in the map
+	for _, q := range b {
+		if _, exists := set[q.ID]; exists {
+			intersection = append(intersection, q)
+		}
+	}
+
+	return intersection
+}
+
+func findQuickstartsByTagsAndDisplayName(tagTypes []models.TagType, tagValues [][]string, displayName string, pagination Pagination) ([]models.Quickstart, error) {
+	var quickstarts []models.Quickstart
+	var quickstartsFilteredByTags []models.Quickstart
+	var quickstartsFilteredByDisplayName []models.Quickstart
+	var err error
+
+	quickstartsFilteredByTags, err = findQuickstartsByTags(tagTypes, tagValues, pagination)
 	if err != nil {
 		return nil, err
 	}
 
-	return quickstarts, nil
+	quickstartsFilteredByDisplayName, err = findQuickstartsByDisplayName(displayName, pagination)
+	quickstarts = intersectQuickstarts(quickstartsFilteredByDisplayName, quickstartsFilteredByTags)
```

Comment:
This is not gonna work because of pagination, if we need combine tags and display name, it all has to happen in one query.